schema.json

```diff
@@ -207,7 +207,9 @@
       "type": "string",
       "description": "A unique identifier for a step, must not resemble a UUID",
       "examples": [ "deploy-staging", "test-integration" ],
-      "pattern": "^(?!^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$).*$"
+      "not": {
+        "pattern": "^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$"
+      }
```

Comment:
Might be worth leveraging https://json-schema.org/understanding-json-schema/reference/string#resource-identifiers

so:
```suggestion
      "not": {
        "format": "uuid"
      }
```